src/payment/strategies/googlepay/googlepay-checkoutcom-initializer.ts

```diff
@@ -0,0 +1,95 @@
+import { round } from 'lodash';
+
+import { Checkout } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import PaymentMethod from '../../payment-method';
+
+import { GooglePaymentData,
+    GooglePayInitializer,
+    GooglePayPaymentDataRequestV2,
+    TokenizePayload } from './googlepay';
+
+export default class GooglePayCheckoutcomInitializer implements GooglePayInitializer {
+
+    initialize(
+        checkout: Checkout,
+        paymentMethod: PaymentMethod,
+        hasShippingAddress: boolean
+    ): Promise<GooglePayPaymentDataRequestV2> {
+        return Promise.resolve(this._mapGooglePayCheckoutcomDataRequestToGooglePayDataRequestV2(
+            checkout,
+            paymentMethod.initializationData,
+            hasShippingAddress
+        ));
+    }
+
+    teardown(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    parseResponse(paymentData: GooglePaymentData): TokenizePayload {
+        try {
+            const token = JSON.parse(paymentData.paymentMethodData.tokenizationData.token);
+            if (!token.signature || !token.protocolVersion || !token.signedMessage) {
+                throw new InvalidArgumentError('Unable to parse response from Google Pay.');
```

Comment:
why do you need a try/catch block when you are basically rethrowing the same thing?